### PR TITLE
Deprecation removal for 6.0 - part I

### PR DESCRIPTION
**Pull request by carldlaird**
## Summary/Motivation:
- This is the first pass at removing old deprecated code from Pyomo. This is partial progress through a list of files that came from grep'ing for deprecations in the code. I have only removed deprecations that are earlier than 5.6.

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by mrmundt — APPROVED** [review]

**Comment by mrmundt** [comment]
@jsiirola @blnicho - Resolved the merge conflicts here.

File: pyomo/core/base/objective.py [hunk 1]
```diff
@@ -350,6 +348,9 @@ def _getitem_when_not_present(self, index):
         if obj is None:
             raise KeyError(index)
         else:
+            # !! THIS SEEMS LIKE A BUG!! - mrmundt
+            # My IDE is yelling about `block` not being defined.
+            # Is it supposed to be Block?
             obj.set_sense(self._init_sense(block, index))
```
Comment (jsiirola) [inline pyomo/core/base/objective.py:354]: This is indeed a bug (and clearly untested).  We should make this:
```python
block = self.parent_block()
obj = self._setitem_when_not_present(
    index, self.rule(block, index))
if obj is None:
    raise KeyError(index)
obj.set_sense(self._init_sense(block, index))
```

File: pyomo/core/tests/unit/test_expression.py [hunk 2]
```diff
@@ -513,7 +513,7 @@ def test_explicit_skip_definition(self):
         model.E[1] = None
         expr = model.E[1]
         self.assertIs(type(expr), _GeneralExpressionData)
-        self.assertIs(expr.value, None)
+        self.assertIs(expr, model.E[1])
```
Comment (jsiirola) [inline pyomo/core/tests/unit/test_expression.py:516]: This is not the correct change:  I think it should be:
```
self.assertIs(expr.expr, None)
```

File: pyomo/core/tests/unit/test_model.py [hunk 3]
```diff
@@ -879,10 +879,9 @@ def c(b):
                           ['y','I','x','c'] )
         self.assertEqual( len(list(EXPR.identify_variables(instance.c.body))), 3 )
 
-    def test_error1(self):
-        model = ConcreteModel()
-        model.x = Var()
-        instance = model.create_instance()
+    def test_error_creating_model_baseclass(self):
+        with self.assertRaises(TypeError):
```
Comment (jsiirola) [inline pyomo/core/tests/unit/test_model.py:883]: It would be better if this were an `assertRaisesRegex` to verify that the "right" `TypeError` was raised.

**Review by jsiirola — CHANGES_REQUESTED** [review]
Only a few minor comments / changes.

**Review by blnicho — APPROVED** [review]

**Review by jsiirola — APPROVED** [review]

**Pull request merged**